ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -804,6 +804,59 @@ public func timer(interval: NSTimeInterval, onScheduler scheduler: DateScheduler
 	}
 }
 
+/// Creates a repeating timer of the variable interval starting with `nextInterval(0)`,
+/// with a reasonable default leeway, sending updates on the given scheduler.
+///
+/// This timer will complete when `nextInterval` returns nil.
```

Comment:
Is there any other good way of completing timer at some point?
I find `timeoutWithError` or manual disposal a bit cumbersome...